### PR TITLE
Tests for SYCL 2020 multi_ptr common constructors

### PR DESCRIPTION
**Pull request by AndreyXRomanov**
Provides tests for multi_ptr common constructors according to SYCL 2020 spec.

Depends on:
- #231
- #237
- #250 

Requires SYCL 2020 multi_ptr class implementation for compilation.

### CODE REVIEW

File: tests/multi_ptr/multi_ptr_common_constructors.h [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+void run_tests(sycl_cts::util::logger &log, const std::string &type_name) {
+  using namespace sycl_cts;
+  using m_ptr_t = sycl::multi_ptr<T, Space, Decorated>;
+  using other_decorated_ptr_t =
+      sycl::multi_ptr<T, Space, sycl::access::decorated::yes>;
+
+  // Kernel name
+  using k_name = multi_ptr_kernel_name<T, Space, Decorated>;
+
+  constexpr size_t types_size = 5;
```
Comment (tomdeakin) [inline tests/multi_ptr/multi_ptr_common_constructors.h:63]: Add a comment as to why this is 5. Same for below.

**Review by tomdeakin — CHANGES_REQUESTED** [review]
A few very minor comments. No check for a deprecated address space - it should exist in SYCL 2020, even though it will eventually go.

File: tests/multi_ptr/multi_ptr_common_constructors.h [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+#include "../common/get_cts_string.h"
+#include "../common/type_list.h"
+
+namespace multi_ptr_common_constructors {
+
+template <typename T, sycl::access::address_space Space,
+          sycl::access::decorated Decorated>
+struct multi_ptr_kernel_name;
+
+/** @brief Provdes text description of test case in case of fail
```
[the remark this answers is not in the document]
Comment (AndreyXRomanov) [reply tests/multi_ptr/multi_ptr_common_constructors.h]: Fixed

File: tests/multi_ptr/multi_ptr_common_constructors.h [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                      "> and type: " + type_name};
+  return message;
+}
+
+template <sycl::access::address_space Space>
+constexpr bool use_local_accessor =
+    (Space == sycl::access::address_space::local_space) ||
+    (Space == sycl::access::address_space::private_space);
+
+/** @brief Provdes verification of multi_ptr common constructors with template
```
[the remark this answers is not in the document]
Comment (AndreyXRomanov) [reply tests/multi_ptr/multi_ptr_common_constructors.h]: Fixed

File: tests/multi_ptr/multi_ptr_common_constructors.h (hunk 1 above)
Comment (AndreyXRomanov) [reply tests/multi_ptr/multi_ptr_common_constructors.h:66]: I agree. Comment added.

File: tests/multi_ptr/multi_ptr_common_constructors.h [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+      same_value_acc[0] = (*(mptr.get()) == *(other.get()));
+    }
+
+    // Check multi_ptr(multi_ptr&&)
+    {
+      m_ptr_t other(ref_acc);
+      auto other_get = *(other.get());
+      m_ptr_t mptr(std::move(other));
+
+      same_type_acc[2] =
```
[the remark this answers is not in the document]
Comment (AndreyXRomanov) [reply tests/multi_ptr/multi_ptr_common_constructors.h]: Good idea. Added variables for indexes.

File: tests/multi_ptr/multi_ptr_common_constructors.h [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    run_tests<T, sycl::access::address_space::generic_space,
+              sycl::access::decorated::yes>(log, type_name);
+    run_tests<T, sycl::access::address_space::global_space,
+              sycl::access::decorated::no>(log, type_name);
+    run_tests<T, sycl::access::address_space::local_space,
+              sycl::access::decorated::no>(log, type_name);
+    run_tests<T, sycl::access::address_space::private_space,
+              sycl::access::decorated::no>(log, type_name);
+    run_tests<T, sycl::access::address_space::generic_space,
+              sycl::access::decorated::no>(log, type_name);
```
[the remark this answers is not in the document]
Comment (AndreyXRomanov) [reply tests/multi_ptr/multi_ptr_common_constructors.h:202]: Added tests with `access::address_space::constant_space`

**Comment by AndreyXRomanov** [comment]
> A few very minor comments. No check for a deprecated address space - it should exist in SYCL 2020, even though it will eventually go.

Thanks for review!

**Review by tomdeakin — APPROVED** [review]
:+1:

**Comment by fraggamuffin** [comment]
LGTM

**Review by keryell — APPROVED** [review]

**Pull request merged**